### PR TITLE
specs for declared(params) inside route_param bug.

### PR DESCRIPTION
**Pull request by Arkanain**
I check some specs on my project and I find out that this functionality work little bit wrong with declared when we have several route_param in one namespace.
For example we have:

```ruby
class Artists
  namespace :artists do
    route_param :id, type: Integer do
      get do
        declared(params)
      end

      params do
        requires :filter, type: String
      end
      get :some_route do
        declared(params)
      end
    end
    route_param :artist_id, type: Integer do
      mount Compositions
    end
  end
end
```
`declared(params)` should return us next hash `{id: 1}`, but it return hash `{id: 1, artist_id: nil}`.
Also in first `route_param` in second endpoint(`get :some_route`) when we call `declared(params)` we have only `filter` params in returned hash `{filter: 'some_filter'}`

P.S.: I find out where we have problem! Problem is in options for route_param. where I use route_param :id without type: Integer everything is ok.

### CODE REVIEW

**Comment by Arkanain** [comment]
@dblock @namusyaka please review this.

**Comment by dblock** [comment]
Well these specs pass... weren't they supposed to fail?

**Comment by Arkanain** [comment]
@dblock It's a regression specs. They represent a problem. Or you want to see a regular specs instead of regression specs?

**Comment by Arkanain** [comment]
@dblock change specs. not it failed and show problem correct.

**Comment by Arkanain** [comment]
Hi guys. Any news about this issue?

**Comment by Arkanain** [comment]
@dblock @namusyaka  Please review my code. I find out where was a problem and fixed it. I really need this fix in my project so if everything ok can you please merge it ASAP.

**Comment by namusyaka** [comment]
Looks good to me.
@dblock Any thoughts on this?

**Comment by dblock** [comment]
Looks good, needs a CHANGELOG entry please.

**Comment by Arkanain** [comment]
@dblock Done.

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 #### Fixes
 
+* [#1430](https://github.com/ruby-grape/grape/pull/1430): Fix for declared(params) inside route_param. - [@Arkanain](https://github.com/Arkanain).
```
Comment (dblock) [inline CHANGELOG.md]: Sorry to be a pain. There shouldn't be a period after `route_param`, also maybe quote the code as below?

**Comment by Arkanain** [comment]
@dblock update.

**Comment by dblock** [comment]
Excellent, merging.

**Pull request merged**